### PR TITLE
GLT-3062: allow changing sample project

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java [hunk 1]
```diff
@@ -274,6 +277,8 @@ public ModelAndView setupForm(@PathVariable Long sampleId, ModelMap model) throw
         .map(Dtos::asDto)
         .collect(Collectors.toList()));
 
+    model.put("projects", mapper.writeValueAsString(projectService.list().stream().map(Dtos::asDto).collect(Collectors.toList())));
```
Comment (apmasell) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java]: Navel gazing question: should we limit to active projects?

**Comment by djcooke** [comment]
Going to add a warning of "This project change will affect X libraries" per sample when changing project on aliquots

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java (hunk 1 above)
Comment (djcooke) [reply miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java]: Yeah, I'll change it to only display active projects + current project if not active

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java [hunk 2]
```diff
@@ -511,7 +517,7 @@ private BulkPropagateSampleBackend(SampleClass targetSampleClass, BoxDto newBox,
     }
 
     @Override
-    protected SampleDto createDtoFromParent(Sample item) {
+    protected SampleDto createDtoFromParent(Sample item) { // TODO: identityConsentLevel
```
Comment (callunity) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java]: Is this TODO still needed? Looks like it's implemented below in 530-532.

Comment (djcooke) [reply miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java]: removed

**Comment by djcooke** [comment]
Added library warning and tests should be fixed now

**Pull request merged**